src/librbd/AioImageRequest.cc

```diff
@@ -279,7 +285,7 @@ void AioImageRead<I>::send_request() {
   auto &image_extents = this->m_image_extents;
   if (image_ctx.object_cacher && image_ctx.readahead_max_bytes > 0 &&
       !(m_op_flags & LIBRADOS_OP_FLAG_FADVISE_RANDOM)) {
-    readahead(get_image_ctx(&image_ctx), image_extents);
+    readahead(get_image_ctx(&image_ctx), image_extents, &this->trace);
```

Comment:
Should readahead be associated with the current trace? This is more like a background process to pre-populate the cache as a result of (potentially) numerous sequential read operations.